common/test/acceptance/tests/video/test_video_events.py

```diff
@@ -0,0 +1,137 @@
+    """ Test video player event emission """
+
+    def test_video_control_events(self):
+        """
+        Scenario: Video component is rendered in the LMS in Youtube mode without HTML5 sources
+        Given the course has a Video component in "Youtube" mode
+        And I play the video
+        And I watch 5 seconds of it
+        And I pause the video
+        Then a "load_video" event is emitted
+        And a "play_video" event is emitted
+        And a "pause_video" event is emitted
+        """
+        load_video_promise = self.create_event_of_type_promise('load_video')
+
+        self.navigate_to_video()
+
+        load_video_event = load_video_promise.fulfill()
+
+        self.assert_payload_contains_ids(load_video_event)
+        assert_event_matches({'event_type': 'load_video'}, load_video_event)
+
+        play_video_promise = self.create_event_of_type_promise('play_video')
+        pause_video_promise = self.create_event_of_type_promise('pause_video')
+
+        self.video.click_player_button('play')
+        self.video.wait_for_position('0:05')
+        self.video.click_player_button('pause')
+
+        play_video_event = play_video_promise.fulfill()
+        self.assert_valid_control_event_at_time(play_video_event, 0)
+
+        pause_video_event = pause_video_promise.fulfill()
+        self.assert_valid_control_event_at_time(pause_video_event, self.video.seconds)
+
+    def assert_payload_contains_ids(self, video_event):
+        """
+        Video events should all contain "id" and "code" attributes in their payload.
+
+        This function asserts that those fields are present and have correct values.
+        """
+        video_descriptors = self.course_fixture.get_nested_xblocks(category='video')
+        video_desc = video_descriptors[0]
+        video_locator = UsageKey.from_string(video_desc.locator)
+
+        expected_event_pattern = {
+            'event': {
+                'id': video_locator.html_id(),
+                'code': '3_yD_cEKoCk'
+            }
+        }
+        assert_event_matches(expected_event_pattern, video_event)
+
+    def assert_valid_control_event_at_time(self, video_event, time_in_seconds):
+        """
+        Video control events should contain valid ID fields and a valid "currentTime" field.
+
+        This function asserts that those fields are present and have correct values.
+        """
+        self.assert_payload_contains_ids(video_event)
+        current_time = json.loads(video_event['event'])['currentTime']
+        self.assertAlmostEqual(current_time, time_in_seconds, delta=0.5)
+
+    def test_strict_event_format(self):
+        """
+        This test makes a very strong assertion about the fields present in events. The goal of it is to ensure that new
+        fields are not added to all events mistakenly. It should be the only existing test that is updated when new top
+        level fields are added to all events.
+        """
```

Comment:
But I don't think that only "level" fields are affected by strict.  I'm concerned that if an extra field is created by a code change in the actual event output, that running with strict=False will not catch these changes.  

And while acceptance tests are more likely to have extra fields, and it's tedious to update them all, I don't see that being the case with unit tests.  I think it would be safer for unit tests to run with strict=True.